### PR TITLE
Camp Fortifications fix

### PR DESCRIPTION
**Pull request by ghost**
#### Summary
```SUMMARY: Bugfixes "Camp Fortifications fix"```

#### Purpose of change
Camp fortifications were not buildable, the companion mission points were cleared before they were read. 
Camp spiked pits were not buildable over the top of trenches, because the strings were not present on the omt id, as update_mapgen changed how that worked.

#### Describe the solution
Moved the companion_mission_points.clear after the check to where to build.
Added  vector of tripoints to bsecamp that lists tripoints of trenches, so the spiked pits know where they can be built.

#### Describe alternatives you've considered
Fortifications in general need a lot of work and changes to how they work and how they are built.
But this at least fixes them to be buildable again.

#### Additional context
N/A

### CODE REVIEW

**Comment by davidpwbrown** [comment]
Not tested serialize commit, will come back to that tomorrow.

**Comment by davidpwbrown** [comment]
Just stuck on trying to fix something else for the moment, I will fix this up hopefully tonight.

File: src/savegame_json.cpp [hunk 1]
```diff
@@ -2987,6 +2987,7 @@ void basecamp::serialize( JsonOut &json ) const
         json.member( "pos", omt_pos );
         json.member( "bb_pos", bb_pos );
         json.member( "expansions" );
+        json.member( "fortifications");
```
Comment (ZhilkinSerg) [inline src/savegame_json.cpp]:
```suggestion
        json.member( "fortifications" );
```

File: src/savegame_json.cpp [hunk 2]
```diff
@@ -2996,6 +2997,11 @@ void basecamp::serialize( JsonOut &json ) const
             json.member( "pos", expansion.second.pos );
             json.end_object();
         }
+        for( const auto &fortification : fortifications ){
```
Comment (ZhilkinSerg) [inline src/savegame_json.cpp]:
```suggestion
        for( const auto &fortification : fortifications ) {
```

**Pull request merged**